daemon/logger/loggerutils/logfile.go

```diff
@@ -232,6 +342,46 @@ func (w *LogFile) openRotatedFiles() (files []*os.File, err error) {
 	return files, nil
 }
 
+func decompressedJsonfile(fileName string, since time.Time) (string, error) {
+	cf, err := os.Open(fileName)
+	if err != nil {
+		return "", err
+	}
+	defer cf.Close()
+
+	rc, err := gzip.NewReader(cf)
+	if err != nil {
+		return "", err
+	}
+	defer rc.Close()
+
+	// Extract the last log entry timestramp from the gzip header
+	extra := &RotateFileMetadata{}
+	err = json.Unmarshal(rc.Header.Extra, extra)
+	if err != nil {
+		return "", err
+	}
+
+	if extra.LastTime.Before(since) {
+		return "", nil
+	}
+
+	tmpFile := fileName + tmpLogfileSuffix
+	rs, err := os.OpenFile(tmpFile, os.O_CREATE|os.O_RDWR, 0640)
+	if err != nil {
+		return "", err
+	}
+	defer rs.Close()
```

Comment:
How come we close here instead of returning the open file?

Reply:
will do.

Reply:
Done.